src/trans.js

```diff
@@ -109,17 +109,16 @@ export default class Trans extends React.Component {
   }
 
   render() {
-    const { children, count, parent } = this.props;
+    const { children, count, parent, i18nKey, ...additionalProps } = this.props;
 
     const defaultValue = nodesToString('', children, 0);
-    const key = this.props.i18nKey || defaultValue;
+    const key = i18nKey || defaultValue;
     const translation = this.t(key, { interpolation: { prefix: '#$?', suffix: '?$#' }, defaultValue, count });
 
-    const additionalProps = {};
     if (this.i18n.options.react && this.i18n.options.react.exposeNamespace) {
       let ns = typeof this.t.ns === 'string' ? this.t.ns : this.t.ns[0];
-      if (this.props.i18nKey && this.i18n.options.nsSeparator && this.props.i18nKey.indexOf(this.i18n.options.nsSeparator) > -1) {
-        const parts = this.props.i18nKey.split(this.i18n.options.nsSeparator);
+      if (rops.i18nKey && this.i18n.options.nsSeparator && this.i18nKey.indexOf(this.i18n.options.nsSeparator) > -1) {
```

Comment:
could you please fix this to be this.props.i18nKey again

Reply:
I can, but I do need to keep the destructuring in place so `i18nKey` does not become part of `additionalProps`, since that will result in a rendering error from React. Since that means i18nKey must be a local variable anyway, I figured that would simplify the spelling here as well.

Reply:
you also can use that destructed i18nKey. Before it just had a typo `rops.i18nKey` and `props.i18nKey`. Anyway will merge now and publish an update.

Reply:
I only noticed those typos later :(. I wonder why they didn't break my testing.